### DISCUSSION TITLE
Plotting Spock on the Tektronix 4014

### DISCUSSION THREAD

**Discussion by GrantMeStrength** [thehighnibble/thehighnibble.github.io#37] [IMSAI8080esp replica]
I saw the picture of Spock plotted on the new TEK: device and thought "that's something I want to do".

The **spock.plt** file can be downloaded as part of this repo: https://github.com/rricharz/make-tek-image

It works with the **plot.com** command that is included on the disk added to the library when you flash the firmware.

One problem: how do you get the **spock.plt** file onto your IMSAI? Here's how I did it, and I welcome an easier way:

1. Install the **cpmtools** suite (details: http://www.moria.de/~michael/cpmtools/) on a Mac I used `brew install cpmtools`
2. Download a floppy disk image from the Library (drag an unmounted disk to the little download arrow)
3. Duplicate the disk image, use `cpmrm` to remove files and copy the **spock.plt** file to it.
4. Realize the **spock.plt** is _TOO LARGE_ to fit onto a floppy so instead, use the 4Mb Hard Drive image file (you are using that, mounted as I: right? It's really convenient.)
5. Edit the **boot.config** from SYS to comment out the hard drive image (if you are using it) and then restart the IMSAI  to unmount it, and then download the image (drag it from the Library to the little download arrow).
6. Add the **spock.plt** file using `cpmcp -f4mb-hd <location of the disk image .hdd> <location of the spock.plt file> 0:spock.plt`
7. Check it's been added with `cpmls <location of the disk image>` 
8. Drag the HDD file back into the IMSAI Library, edit **boot.config** to mount the hard drive image, reboot the IMSAI.
9. Now the file is on I: and you can use **plot.com** to draw it. Have a cup of tea as it renders. Maybe two!

**Comment by villaromba** [comment]
I used Xmodem (XM) via Teraterm (Win10) straight on to the hard disk. You can also create your own images. - see the 'show and tell' on the menu bar. If you are on linux then ...................................................!!!!!!!!!!!!

**Reply by GrantMeStrength** [reply]
You have a serial link between the IMSAI and your computer? Or does this work over Wifi?

**Reply by villaromba** [reply]
I use the USB connection from Imsai to PC (not via the GUI) . This then shows up on Teraterm . This is as per the  Manual : Serial UART over USB. The ESP32-PICO-KIT supports serial communications from UART0 over USB. It uses a Silicon Labs CP210x USB to UART bridge
1/ connect the ESPP32-PICO-KIT to a PC using a suitable USB cable
2/ start a terminal emulator on the PC set for 115200 baud 8N1 connected to the serial device your OS identifies the ESP32-PICO-KIT on
Windows will be a COMx: port 

I have my port set to 9600 baud as XM seems more reliable (in my case) at a lower speed.

**Comment by dmcnaugh** [comment]
### RE: Step 6.
There is also a format in **cpmtools** called `z80pack-hd` and this is the one @udo-munk recommends on his site for Z80PACK
The two defs. are different so I'd use `z80pack-hd` just to be sure.
